modules/ort/src/main/java/org/eclipse/sw360/antenna/ort/resolver/OrtResultArtifactResolver.java

```diff
@@ -0,0 +1,154 @@
+
+    public OrtResultArtifactResolver(OrtResult result) {
+        licenseFindings = result.collectLicenseFindings(false);
+    }
+
+    @Override
+    public Artifact apply(Package pkg) {
+        Artifact a = new Artifact("OrtResult")
+                .addFact(new ArtifactMatchingMetadata(MatchState.EXACT));
+
+        mapCoordinates(pkg).ifPresent(a::addFact);
+        mapSourceUrl(pkg).ifPresent(a::addFact);
+
+        mapDeclaredLicense(pkg).ifPresent(a::addFact);
+        mapFilename(pkg).ifPresent(a::addFact);
+        mapHomepage(pkg).ifPresent(a::addFact);
+
+        mapObservedLicense(pkg).ifPresent(a::addFact);
+        mapCopyrights(pkg).ifPresent(a::addFact);
+
+        return a;
+    }
+
+    private static Optional<ArtifactCoordinates> mapCoordinates(Package pkg) {
+        String namespace = pkg.getId().getNamespace();
+        String name = pkg.getId().getName();
+        String version = pkg.getId().getVersion();
+
+        switch (pkg.getId().getType().toLowerCase()) {
+            case "nuget":
+            case "dotnet":
+                return Optional.of(mapDotNetCoordinates(name, version));
+            case "maven":
+                return Optional.of(mapMavenCoordinates(namespace, name, version));
+            case "npm":
+                return Optional.of(mapJavaScriptCoordinates(name, version));
+            default:
+                return Optional.of(mapSimpleCoordinates(name, version));
+        }
+    }
+
+    private static ArtifactCoordinates mapDotNetCoordinates(String name, String version) {
+        return new DotNetCoordinates.DotNetCoordinatesBuilder()
+                .setPackageId(name)
+                .setVersion(version)
+                .build();
+    }
+
+    private static ArtifactCoordinates mapMavenCoordinates(String namespace, String name, String version) {
+        return new MavenCoordinates.MavenCoordinatesBuilder()
+                .setGroupId(namespace)
+                .setVersion(version)
+                .setArtifactId(name)
+                .build();
+    }
+
+    private static ArtifactCoordinates mapJavaScriptCoordinates(String name, String version) {
+        return new JavaScriptCoordinates.JavaScriptCoordinatesBuilder()
+                .setName(name)
+                .setVersion(version)
+                .setArtifactId(name + "-" + version)
+                .build();
+    }
+
+    private static ArtifactCoordinates mapSimpleCoordinates(String name, String version) {
+        return new GenericArtifactCoordinates(name, version);
+    }
+
+    private static Optional<ArtifactSourceUrl> mapSourceUrl(Package pkg) {
+        final Stream<String> urls = Stream.of(pkg.getVcsProcessed().getUrl(), pkg.getSourceArtifact().getUrl());
+
+        return urls.filter(u -> !u.isEmpty())
+                .findFirst()
+                .map(ArtifactSourceUrl::new);
+    }
+
+    private static Optional<DeclaredLicenseInformation> mapDeclaredLicense(Package pkg) {
```

Comment:
Differently because this is straight forward, more or less, I understand how declared licenses are handled